src/views/Drawer.tsx

```diff
@@ -0,0 +1,553 @@
+    this.containerWidth.setValue(e.nativeEvent.layout.width);
+
+  private handleDrawerLayout = (e: LayoutChangeEvent) => {
+    this.drawerWidth.setValue(e.nativeEvent.layout.width);
+    this.toggleDrawer(this.props.open);
+
+    // Until layout is available, drawer is hidden with opacity: 0 by default
+    // Show it in the next frame when layout is available
+    // If we don't delay it until the next frame, there's a visible flicker
+    requestAnimationFrame(() => this.drawerOpacity.setValue(1));
+  };
+
+  private toggleDrawer = (open: boolean) =>
+    this.nextIsOpen.setValue(open ? TRUE : FALSE);
+
+  private toggleStatusBar = (hidden: boolean) => {
+    const { hideStatusBar, statusBarAnimation } = this.props;
+
+    if (hideStatusBar && this.isStatusBarHidden !== hidden) {
+      this.isStatusBarHidden = hidden;
+      StatusBar.setHidden(hidden, statusBarAnimation);
+    }
+  };
+
+  render() {
+    const {
+      open,
+      locked,
+      drawerPosition,
+      drawerType,
+      swipeEdgeWidth,
+      contentContainerStyle,
+      drawerStyle,
+      overlayStyle,
+      onGestureRef,
+      renderDrawerContent,
+      renderSceneContent,
+    } = this.props;
+
+    const right = drawerPosition === 'right';
+
+    const contentTranslateX = drawerType === 'front' ? 0 : this.translateX;
+    const drawerTranslateX =
+      drawerType === 'back'
+        ? I18nManager.isRTL
+          ? multiply(this.drawerWidth, DIRECTION_RIGHT)
+          : this.drawerWidth
+        : this.translateX;
+
+    const offset = I18nManager.isRTL ? '100%' : multiply(this.drawerWidth, -1);
+
+    // TODO: hitSlop is broken when on Android when drawer is on right
+    const hitSlop = right
+      ? // Extend hitSlop to the side of the screen when drawer is closed
+        // This lets the user drag the drawer from the side of the screen
+        // @ts-ignore
+        { right: 0, width: open ? undefined : swipeEdgeWidth }
+      : { left: 0, width: open ? undefined : swipeEdgeWidth };
+
+    return (
+      <PanGestureHandler
+        ref={onGestureRef}
+        activeOffsetX={[-SWIPE_DISTANCE_MINIMUM, SWIPE_DISTANCE_MINIMUM]}
+        failOffsetY={[-SWIPE_DISTANCE_MINIMUM, SWIPE_DISTANCE_MINIMUM]}
+        onGestureEvent={this.handleGestureEvent}
+        onHandlerStateChange={this.handleGestureEvent}
+        hitSlop={hitSlop}
+        enabled={!locked}
+      >
+        <Animated.View
+          onLayout={this.handleContainerLayout}
+          style={styles.main}
+        >
+          <Animated.View
+            style={[
+              styles.content,
+              {
+                transform: [{ translateX: contentTranslateX }],
+              },
+              contentContainerStyle as any,
+            ]}
+          >
+            {renderSceneContent({ progress: this.progress })}
+            <TapGestureHandler onHandlerStateChange={this.handleTapStateChange}>
+              <Animated.View
+                pointerEvents={open ? 'auto' : 'none'}
```

Comment:
i believe that this should be controlled by the animated value so that it is only `none` when the drawer is fully closed (or effectively fully closed, maybe 0 +- some epsilon)